File: src/1389-Create-Target-Array-in-the-Given-Order/1389.cpp
class Solution {
public:
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
        vector<int> res;
        for (int i = 0; i < nums.size(); i++)
            res.insert(res.begin() + index[i], nums[i]);
        return res;
    }
};

File: src/1389-Create-Target-Array-in-the-Given-Order/1389.java
class Solution {
    public int[] createTargetArray(int[] nums, int[] index) {
        List<Integer> data = new ArrayList<>(nums.length);
        for (int i = 0; i < nums.length; i++) {
            data.add(index[i], nums[i]);
        }
        
        int[] res = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            res[i] = data.get(i);
        }
        return res;
    }
}